src/app/components/play-round/play-round.component.html

```diff
@@ -17,13 +17,18 @@ <h1 class="nest-count">{{(this.studentRoundService.currentNestPollen$ | async) }
   [ngStyle]="{'transform': 'translate(-50%, 25%) scale(' + calculateBeeScale(bee.relative_size) + ')'}"
   [alt]="bee.name"
 >
-<ng-container *ngIf="{ count: (this.studentRoundService.currentBeePollen$ | async) } as currentBeePollen">
+<ng-container
+  *ngIf="{
+    currentBeePollen: (studentRoundService.currentBeePollen$ | async),
+    recentFlowerInteractions: (studentRoundService.recentFlowerInteractions$ | async)
+  } as context"
+>
   <button
     mat-raised-button
     class="interact-button"
     color="primary"
     *ngIf="this.foundRoundMarker$ | async as marker"
-    [disabled]="!marker.active || currentBeePollen.count >= 3 && !marker.isNest"
+    [disabled]="!canVisit(marker, context)"
```

Comment:
We really need to avoid using a function call in data binding like this. This function will be called *every* change detection tick and either tank performance or just hang.

Reply:
Ah, good to know, thank you very much! Looks like we can fix that problem by using [a pipe instead of a function call](https://medium.com/showpad-engineering/why-you-should-never-use-function-calls-in-angular-template-expressions-e1a50f9c0496#72eb), since Angular is smarter about when to change-detect a pipe. I'll try to implement that later today! :)

Reply:
(Actually, we talked about this at the meeting, and we're probably going to put this information on a field on the `RoundMarker`s!)

Reply:
Since that'll involve a bit of refactoring, I probably get to that some other day :)